### PR TITLE
Metrics: Add APICast status.

### PR DESCRIPTION
**Pull request by eloycoto**
This change adds a new prometheus metric, that it's apicast_status. The
main reasons to add this new metric are:

- If the upstream/client close the connection, the upstream_status will
not be reported at all.
- If upstream API is not available, the 502 that APICast send to the
user will never be on a metric.
- If auth invalid, rate limit, the status will never be reported.
- If upstream reponse is cached, the upstream_status will never be
updated. (@todo, added a metric for cached responses?)
- If service cannot be found, the status will never be found.

Signed-off-by: Eloy Coto <eloy.coto@acalustra.com>

### CODE REVIEW

File: doc/prometheus-metrics.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 | Metric                             | Description                                                      | Type      | Labels                                                       | Policy         |
 |------------------------------------|------------------------------------------------------------------|-----------|--------------------------------------------------------------|----------------|
+| apicast_status                     | Number of response status send by APIcast to client              | counter   | -                                                            | Default        |
```
Comment (davidor) [inline doc/prometheus-metrics.md:5]: We should add `status` in the `labels` column.

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Added Nginx request_id variable to context [PR #1184](https://github.com/3scale/APIcast/pull/1184)
 - Added HTTP verb on url_rewriten [PR #1187](https://github.com/3scale/APIcast/pull/1187)  [THREESCALE-5259](https://issues.jboss.org/browse/THREESCALE-5259)
 - Added custom_metrics policy [PR #1188](https://github.com/3scale/APIcast/pull/1188) [THREESCALE-5098](https://issues.jboss.org/browse/THREESCALE-5098)
+- New apicast_status Prometheus metric [THREESCALE-5417](https://issues.jboss.org/browse/THREESCALE-5417) [PR #1200](https://github.com/3scale/APIcast/pull/1200)
```
Comment (porueesq) [inline CHANGELOG.md:26]: Added a Prometheus metric called ` apicast_status` [THREESCALE-5417](https://issues.jboss.org/browse/THREESCALE-5417) [PR #1200](https://github.com/3scale/APIcast/pull/1200)

File: t/prometheus-metrics.t [hunk 3]
```diff
@@ -424,3 +424,70 @@ qr/upstream_status\{status="200",service_id="21",service_system_name="bar"\} 1/
 ]]
 --- no_error_log
 [error]
+
+
+=== TEST 8: APICast status metric
+This metric report APICast status, because service cannot be found, auth is
```
Comment (porueesq) [inline t/prometheus-metrics.t:430]: This metric reports APIcast status in the following cases: when the service cannot be found, auth is

File: t/prometheus-metrics.t [hunk 4]
```diff
@@ -424,3 +424,70 @@ qr/upstream_status\{status="200",service_id="21",service_system_name="bar"\} 1/
...
+invalid, the response cache from upstream is cached, or upstream is not
+available, so upstream_status is not a valid metric at all, to monitor APICast
```
Comment (porueesq) [inline t/prometheus-metrics.t:432]: available. `upstream_status` is not a valid metric to monitor APICast

File: t/prometheus-metrics.t [hunk 5]
```diff
@@ -424,3 +424,70 @@ qr/upstream_status\{status="200",service_id="21",service_system_name="bar"\} 1/
 ]]
 --- no_error_log
 [error]
+
+
+=== TEST 8: APICast status metric
+This metric report APICast status, because service cannot be found, auth is
+invalid, the response cache from upstream is cached, or upstream is not
```
Comment (porueesq) [inline t/prometheus-metrics.t:431]: not valid, the response cache from upstream is cached, or the upstream is not

**Pull request merged**